go/interpreter/lfvm/memory_test.go

```diff
@@ -47,3 +52,429 @@ func TestExpansionCosts(t *testing.T) {
+			}
+		})
+	}
+}
+
+func TestMemory_SetByte_SuccessfulCases(t *testing.T) {
+
+	baseData := []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef}
+
+	tests := map[string]struct {
+		memory []byte
+		offset uint64
+	}{
+		"regular empty memory": {},
+		"regular with memory": {
+			memory: baseData,
+			offset: 0},
+		"regular with memory and offset": {
+			memory: baseData,
+			offset: 4},
+		"offset trigger memory expansion": {
+			memory: baseData[:4],
+			offset: 8},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			m.store = test.memory
+			ctxt.gas = tosca.Gas(100)
+			value := byte(0x12)
+
+			err := m.SetByte(test.offset, value, &ctxt)
+			if err != nil {
+				t.Errorf("unexpected error, want: %v, got: %v", nil, err)
+			}
+			if m.Len() < test.offset {
+				t.Errorf("unexpected memory size, want: %d, got: %d", test.offset, m.Len())
+			}
+			if m.store[test.offset] != value {
+				t.Errorf("unexpected value, want: %v, got: %v", value, m.store[test.offset])
+			}
+		})
+	}
+}
+
+func TestMemory_SetByte_ErrorCases(t *testing.T) {
+
+	tests := map[string]struct {
+		offset   uint64
+		gas      tosca.Gas
+		expected error
+	}{
+		"not enough gas": {
+			offset:   64,
+			gas:      0,
+			expected: errOutOfGas},
+		"offset overflow": {
+			offset:   math.MaxUint64,
+			gas:      100,
+			expected: errGasUintOverflow},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			m.store = []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef}
+			ctxt.gas = test.gas
+
+			err := m.SetByte(test.offset, 0x12, &ctxt)
+			if !errors.Is(err, test.expected) {
+				t.Errorf("unexpected error, want: %v, got: %v", test.expected, err)
+			}
+		})
+	}
+}
+
+func TestMemory_SetWord_ReportsNotEnoughGas(t *testing.T) {
+
+	testValue := []byte{
+		0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+		0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F,
+		0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17,
+		0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F}
+
+	tests := map[string]struct {
+		memory        []byte
+		gas           tosca.Gas
+		offset        uint64
+		expectedData  []byte
+		expectedError error
+	}{
+		"regular": {
+			gas:          3,
+			expectedData: testValue},
+		"not enough gas": {
+			gas:           1,
+			offset:        32,
+			expectedError: errOutOfGas},
+		"offset fits in memory": {
+			gas:          6,
+			offset:       24,
+			expectedData: testValue},
+		"offset same as memory size": {
+			memory:       append(testValue, testValue...),
+			gas:          3,
+			expectedData: testValue,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			ctxt.gas = test.gas
+			err := m.SetWord(test.offset, uint256.NewInt(1).SetBytes(test.expectedData), &ctxt)
```

Comment:
You keep initializing an `uint256.Int` with a value of 0 or 1 for no reason. A cleaner way of handing a situation where you need an instance but the value is irrelevant could look like this:
```suggestion
			err := m.SetWord(test.offset, new(uint256.Int).SetBytes(test.expectedData), &ctxt)
```